standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -593,8 +593,8 @@ struct Table {
   22: optional byte accessType,
   23: optional list<string> requiredReadCapabilities,
   24: optional list<string> requiredWriteCapabilities
-  25: optional i64 id,                 // id of the table. It will be ignored if set. It's only for
-                                        // read purposed
+  25: optional i64 id,                 // id of the table. It will be ignored if set. It's only for read purposes
+  26: optional i64 txnId,              // txnId associated with the table creation
```

Comment:
Quick question: Why not do the whole stuff on client side, and without the HMS API change?

Reply:
@pvary, could you please elaborate on that. We can't just override tableName if that's what you are thinking.

Reply:
> @pvary, could you please elaborate on that. We can't just override tableName if that's what you are thinking.

Why not just override the tableName?

Reply:
AFAIK, in multiple places, we are referencing the table by its name and not id, like when checking if a table exists.

Reply:
Sorry. My mistake. Instead of:

> Why not just override the tableName?

I wanted to ask:

Why not just override the table location?

Reply:
That could work, but it would scatter table path generation logic.